### DISCUSSION TITLE
Support VitePress framework

### DISCUSSION THREAD

**Discussion by camilogarciabotero** [quarto-dev/quarto-cli#9292] [Feature Requests]
### Description

Hey all,

I wonder if there is any chance to support [VitePress](https://vitepress.dev) framework. Quarto currently supports similar Markdown [SSG frameworks](https://quarto.org/docs/output-formats/all-formats.html#markdown). But VitePress seems like a very popular one and there is even a possible adoption as a default framework for the [Julia programming language documentation](https://lazarusa.github.io/JuliaDocs/) see the official [PR](https://github.com/JuliaLang/julia/pull/53988) based on [DocumenterVitepress.jl](https://github.com/LuxDL/DocumenterVitepress.jl) (the new package that is supporting the framework in Julia documentation). I want to highlight that Quarto and Julia will, [hopefully](https://github.com/quarto-dev/quarto-cli/pull/8645), be used smoothly in the future. Moreover, the JavaScript framework behind VitePress, `Vue.js`, also comes with a bunch of features that, I guess, could be useful to create new Quarto filters, themes, and extensions. In summary three reasons to adopt VitePress format framework.

1. Very popular 
2. Could enable nice integration between Quarto and Julia
3. Open up more flexibility to Quarto

Best.

**Comment by cameronraysmith** [comment]
I agree this would be a very nice feature.
Vitepress almost works with the built-in [github-flavored markdown export](https://quarto.org/docs/reference/formats/markdown/gfm.html) as this is [pretty well-supported by vitepress](https://vitepress.dev/guide/markdown).
It would probably require some additional work to automatically generate the correct [configuration](https://vitepress.dev/reference/site-config) such as the nav and sidebar and [enable markdown-it mathjax support](https://vitepress.dev/guide/markdown#math-equations) by default.

---

One way of achieving this in a manner that could broaden overall support for SSGs could be to interface with typescript structured content management tools. See #10394 and #10922.
